packages/cycle-scripts-one-fits-all/template/src/index.ts

```diff
@@ -1,37 +1,29 @@
-import xs from 'xstream';
 import { setup, run } from '@cycle/run';
-import { restartable, rerunner } from 'cycle-restart';
-import { makeDOMDriver } from '@cycle/dom';
-import { makeHTTPDriver } from '@cycle/http';
-import { timeDriver } from '@cycle/time';
+import { rerunner } from 'cycle-restart';
 import isolate from '@cycle/isolate';
 import onionify from 'cycle-onionify';
+import storageify from 'cycle-storageify';
 
-import { Component, Sources, RootSinks } from './interfaces';
+import { mkDrivers, Component } from './drivers';
 import { App } from './app';
 
-const main : Component = onionify(App);
-
-let drivers : any, driverFn : any;
-/// #if PRODUCTION
-drivers = {
-    DOM: makeDOMDriver('#app'),
-    HTTP: makeHTTPDriver(),
-    Time: timeDriver
+// TODO PR upstream - 2nd,3rd options should be optional
+declare type StorageifyOptions = {
+    key : string;
+    serialize(state : any): string;
+    deserialize(stateStr : string): any;
 };
-/// #else
-driverFn = () => ({
-    DOM: restartable(makeDOMDriver('#app'), { pauseSinksWhileReplaying: false }),
-    HTTP: restartable(makeHTTPDriver()),
-    Time: timeDriver
-});
-/// #endif
-export const driverNames : string[] = Object.keys(drivers || driverFn());
```

Comment:
Yes, that is possible, but unlikely. Anyway I have a better idea:
```ts
const driverThunks = {
    DOM: () => makeDOMDriver('#app'),
    HTTP: () => makeHTTPDriver(),
    Time: () => timeDriver,
    router: () => makeRouterDriver(
        createBrowserHistory(),
        switchPath as RouteMatcher
    ),
    storage: () => storageDriver,
    speech: () => speechDriver
};
const driverNames = Object.keys(driverThunks).concat(['onion']);

const mapReducePairs = fn => Object.keys(driverThunks)
    .map(k => [k, driverThunks[k] ])
    .map(fn)
    .map((k, v) => ({ [k]: v }))
    .reduce(Object.assign, {});

/// #idef PRODUCTION
run(main, mapReducePairs(([k, v]) => [k, v()]))
/// #else
const mkDrivers = () => mapReducePairs(([k, v]) => {
    if(k === 'DOM') { return [k, restartable(v(), { pauseSinksWhileReplaying: false })]; }
    if(k === 'Time') { return [k, v()]; }
    return [k, restartable(v())];
});
const rerun = rerunner(setup, mkDrivers, isolate);
```
I would keep all `ifdef` stuff in one file (the index file)